### PR TITLE
demo/remote: add GCP Managed Instance group demo.

### PR DESCRIPTION
**Pull request by jrasell**
Co-authored-by: Luiz Aoqui <luiz@hashicorp.com>

### CODE REVIEW

File: demo/remote/gcp/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Google Cloud Platform
+In order to setup some basic GCP resources for the demo, we will utilise the
```
Comment (cgbaker) [inline demo/remote/gcp/README.md:2]:
```suggestion
In order to set up some basic GCP resources for the demo, we will utilise the
```

File: demo/remote/gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+number of Nomad jobs which provide metrics and dashboarding as well as a demo application and
+routing provided by Traefik. It may take a few seconds for all the applications to start, so if
+any of the URLs doesn't load the first time, wait a little bit and give it a try again.
+
+Please also copy the export commands and run these in the terminal where the rest of the demo will
+be run.
+
+The application is pre-configured with a scaling policy, you can view this by opening the job file
+or calling the Nomad API. The application scales based on the average number of active connections,
+and we are targeting an average of 10 per instance of our app.
```
Comment (cgbaker) [inline demo/remote/gcp/README.md:107]:
```suggestion
and we are targeting an average of 10 connections per instance of our app.
```

File: demo/remote/gcp/terraform/control/.terraform.lock.hcl [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# This file is maintained automatically by "terraform init".
```
Comment (cgbaker) [inline demo/remote/gcp/terraform/control/.terraform.lock.hcl:1]: should this be in .gitignore?

Comment (cgbaker) [reply demo/remote/gcp/terraform/control/.terraform.lock.hcl:1]: huh, apparently so: https://www.terraform.io/docs/language/dependency-lock.html
> Terraform has made some changes to the provider dependency selections recorded
> in the .terraform.lock.hcl file. Review those changes and commit them to your
> version control system if they represent changes you intended to make.

TIL.

**Review by cgbaker — APPROVED** [review]
👍 🏆

File: demo/remote/gcp/terraform/control/.terraform.lock.hcl (hunk 3 above)
Comment (lgfa29) [reply demo/remote/gcp/terraform/control/.terraform.lock.hcl:1]: So it should _not_ be in .gitignore (i.e. having it in the repo is the right thing to do)?

Comment (cgbaker) [reply demo/remote/gcp/terraform/control/.terraform.lock.hcl:1]: You are correct.

**Review by unrealcloud — APPROVED** [review]

**Pull request merged**